src/ucp/core/ucp_worker.c

```diff
@@ -265,12 +265,10 @@ static ucs_status_t ucp_worker_wakeup_init(ucp_worker_h worker,
         goto out;
     }
 
-    if (params->field_mask & UCP_WORKER_PARAM_FIELD_EVENTS) {
-        events = params->events;
-    } else {
-        events = UCP_WAKEUP_RMA | UCP_WAKEUP_AMO | UCP_WAKEUP_TAG_SEND |
-                 UCP_WAKEUP_TAG_RECV | UCP_WAKEUP_TX | UCP_WAKEUP_RX;
-    }
+    events = UCP_PARAM_VALUE(WORKER, params, events, EVENTS,
+                             UCP_WAKEUP_RMA | UCP_WAKEUP_AMO |
+                                     UCP_WAKEUP_TAG_SEND | UCP_WAKEUP_TAG_RECV |
+                                     UCP_WAKEUP_TX | UCP_WAKEUP_RX);
```

Comment:
maybe define UCP_WAKEUP_ALL?

Reply:
Is it ok that UCP_WAKEUP_ALL will not include UCP_WAKEUP_EDGE?
If so this will be an API change. I think it should be in another PR, shouldn't it?

Reply:
ok, didn't notice it's API